torch/tensor.py

```diff
@@ -504,10 +504,13 @@ def __cuda_array_interface__(self):
         itemsize = self.storage().element_size()
 
         shape = tuple(self.shape)
-        strides = tuple(s * itemsize for s in self.stride())
+        if self.is_contiguous():
```

Comment:
it would be nice to comment whether this is an optimization or part of the protocol, but not strictly necessary.

Reply:
Added a comment based on my current understanding, but it looks discussions are still going at https://github.com/numba/numba/issues/4175, so I will adapt it after all reaching a consensus :).